Add new design - Vocaluxe 2024

marwin89 opened the pull request:
### Summary
@flokuep: This PR adds a totally new design called Vocaluxe 2024 made by Marwin89. All elements are checked to be open source compatible. Please read the full license document in themes folder.

### What does this PR do?

- add a totally new theme "Vocaluxe 2024" made by Marwin89
- incl. new bgm, small avatar pack, splash logo, re-adjust font-outline, six skins (standard, blue, red, green, dark mode, black), etc.
- Everything in this theme (incl. bgm and logo-font) is free for commercial use without attribution (most based on pixabay content and pexels).
- All copyright and license information are included in the license document in themes folder.
- This Design is based on Genius Theme (from latest nightly build 2023-11-23).

### Motivation

- motivate new game programmers to join
- motivate designers and new game artists to join
- total re-design, resurgence of interesst
- broaden the audience for this game
- new standard-theme after 5 years
- new basis for upcoming themes

### Closes Issue(s)

- #602 

### Additional Notes
- I tested the new theme as good as possible and didn't find a failure. But please do a short check.
- The source elements are in Full HD 1920x1080 available, I downsized it to HD ready (1280x720) for vocaluxe. I could make higher resolution if vocaluxe supports it in future.
- we can have genius theme in a seperate repository, where it can be maintained by interessted people.
- Theme fits to recommended [style guidelines of open music games.org](https://www.open-music-games.org/community/meet-the-teams#guide-for-artists)

### Screenshots / Video
Check out [Demo Video](https://drive.google.com/file/d/1SBfFXP3NU25pkDkuDdrZvW_wZKMtTr32/view?usp=sharing) 

![vocaluxe-2024-1-loading-screen](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/abcc1001-dcd3-480f-b6a8-66c9cf99b050)
![vocaluxe-2024-2-main-screen](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/2854e5b6-5d45-482c-b71a-02fbc51b3d8f)
![vocaluxe-2024-3-sub-screen](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/69b55e28-538d-4faf-a2e1-f2da9edeaa46)
![vocaluxe-2024-4-song-screen](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/703679e6-9378-4a4e-8193-fc8c74b9e864)
![vocaluxe-2024-5-sing-screen](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/88a83aaa-3b1b-440c-9e73-ef98e67ed20c)
![vocaluxe-2024-6-result-screen](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/381ea4c7-1587-47f6-8261-f4a26e344337)
![vocaluxe-2024-9-blue](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/02b3550d-12a1-47b8-84ad-71b51b9e689c)
![vocaluxe-2024-10-Red](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/cff225f5-8e7a-4e3b-99eb-bc34d41af257)
![vocaluxe-2024-11-black](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/90ddbbd7-fcfa-4d91-bc7e-f55a97cd5c4b)
![vocaluxe-2024-11-dark-mode](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/f07f3ed8-4542-44ca-a00a-8502c4df54a0)
![vocaluxe-2024-11-green](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/72d8ad42-7df4-4151-bb9a-ff4361c16808)

marwin89 committed:
Add new design - Vocaluxe 2024

This PR adds a totally new design called Vocaluxe 2024 made by Marwin89. All elements are checked to be open source compatible. Please read the full license document.

flokuep commented on Output/Fonts/Fonts---.xml:
```diff
@@ -0,0 +1,35 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
```
That filename is strange? It seems to be a some kind of different version than the other Fonts.xml file. Is that intendend?

marwin89 replied:
This was a backup file for the former font configuration of genius theme. I made the outline more subtile and a little bit transparent. I thought to keep it (but we can move this to a genius theme repository). I removed it here. It's just a small detail.

flokuep commented on Output/Profiles/guest1.xml:
You've talked about some new profile pictures in the PR description. Are they missing? IMHO we should have at least three guest-profiles in place, so that there is an easy start-up configuration.

marwin89 replied:
oh I forgot to add them.
In commit 2: I added the pictures and added some new guest avatars (could still be better, but it's a progress). In future I will find a proper solution (for performous I already made a huge official avatarpack sucessfully). 
yes, I agree let's keep the guest profiles.

flokuep reviewed (commented):
Thank your very much for providing a new theme! I really like it :)

There are some minor things (beside the comments in the changes itself), I'd like to discuss before merging this pr:
- Adding theme-elements is a big effort while implementing new features, from my perspective, we should just maintain one theme within Vocaluxe directly, so if we go with this one, we should remove the other one (or move it to a new repository so that it's possible to be maintained by other people).
- I really don't like the VOCALUXE-logo on the sceen. It's some kind of boring. Maybe a different font would be a better match there? Something more elegant, or even writing VOCALUXE in small caps could help?
- There is also some theme-stuff within the party-modes. I guess they need to be adjusted also?

marwin89 commented:
@flokuep : Thanks for your feedback 😃 

### 1 New Standard Theme:
Yes, I agree, we should move Genius Theme to a seperate repository and make vocaluxe 2024 the maintained standard for new features. That's the easiest way to safe effort.

### 2 Party Mode Theming:
I will check them, (as well as the comments)

### 3 Logo:
I understand what you mean. I try to make it more interessting. Which following variants do you like the most? 1 / 2 / 3

Nr: 1
![vocaluxe-2024-cygre-extrabold-italic-year](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/3710cc1c-60d2-4976-8bf7-7460200d7ebe)
Nr: 2
![vocaluxe-2024-cygre-extrabold-italic-year2](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/e5e2e14e-d89d-4238-a342-2a39bf0f99fd)
Nr: 3
![vocaluxe-2024-cygre-extrabold-italic-year3](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/4a3355d3-a6eb-4c35-a2c8-6200952717ab)

marwin89 committed:
Small corrections

remove duplicated font---.xml file, added new guest avatars (transparent) and 4 player avatars

marwin89 commented:
### 2 Party Mode Theming ✅
- the two Party modes are working fine with the new theme.
- (It seems I just can't rename the "default"-Theme folders in the "PartyModes"-Folder. When I do that the two modes don't work at all. I guess it's fine for now.)

so only the logo topic is open.
I guess we should take a little time here. Maybe one or two feedback rounds till it's more satisfying. (try to get some feedback from designer collegues)

flokuep commented:
Thanks for checking and resolving my comments :)

I'd prefer variant 1, but I am still not sure about the 2024 label. It looks like a version number or a subtitle of vocaluxe and could be misleading?

Taking some more design rounds is a good idea!

jimmyhawkin commented:
Is say version 1 with s big V. Allso is 2024 needed?

daggeg commented:
Nr 1 is the best by far in my opinion. And as the others say drop the 2024 :)

daggeg commented:
Oh, I forgot. Would it be possible to add a green theme? For instance one that goes from light to dark green?

marwin89 commented:
### 3 Logo (2nd round)
- thanks for your feedback. so here is the second round for creating a new vocaluxe-logo.
- Six variations based on former variant 1. year is removed
- What do you like? I'd prefer 4 and 1 currently. then 3 and 2.
![vocaluxe-2024-logo-developement-variations](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/4bc7f438-748f-4840-930d-3384c39e790b)

### 4 Skin variations
- @daggeg . A green variation is a good idea. 👋 I added one ✅ 
- plus: a dark mode and also a black theme (for the metal, rock and underground people). ✅ 
- So now we have six color-variants in total. A good foundation. We can have more in another iteration.
![vocaluxe-2024-11-green](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/07417504-0ca8-46b1-b9ed-c1c1a53bb7a0)
![vocaluxe-2024-11-black](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/7be697e1-0964-43af-8236-8c1824552e04)

flokuep commented:
My vote goes for number 1.

Thank you for creating the additional colours.

marwin89 committed:
implement new logo and more

implement new logo, add color backgrounds, rework textures

flokuep commented:
@marwin89 fyi: I've just merged the basic rap support and enhanced pause screen prs some minutes ago. You need to adapt the theme related changes now for the new theme 🙃

marwin89 committed:
implement logo in web-app

implement logo in web-app header, re-coloring dividers in song library

marwin89 committed:
Update license and changelogs

Update license file and moved license file to main license folder. removed duplicated license txt. Update changelogs.

Stefan1200-de commented:
> Is say version 1 with s big V. Allso is 2024 needed?

Yeah, I would also say with a big V (also without 2024).
About the second round of logos: I prefer the second one (maybe again with a big V?).

Just an idea:
But I'm also interested in seeing a variant with a normal "o" but with a microphone instead of the "L".
Maybe this license free image, it should be easy to change the color, size and rotation to fit into the font somehow?
![image](https://github.com/Vocaluxe/Vocaluxe/assets/15168351/d1b4536b-7eda-473d-b7fa-5b808fda2df4)
Source: https://openclipart.org/detail/6631/simple-microphone

marwin89 committed:
Integrate rap note feature

- Integrate rap note feature in vocaluxe 2024 design
- add new rap icon
- integrate change: show song name in sing screen pause

marwin89 committed:
Merge branch 'develop' into vocaluxe-2024-design-marwin89

marwin89 commented:
> fyi: I've just merged the basic rap support and enhanced pause screen prs some minutes ago. You need to adapt the theme related changes now for the new theme 🙃

@flokuep:
- a) very nice 😃 I adapted rap support (and changes on pause screen) for vocaluxe 2024 design.
- b) My designer friends approved the logo (var1) yesterday. We should go for that (for now) (updates always possible) ✅ 
- c) I implemented new logo and updated textures in theme
- d) new logo is also in web-app now
- e) I updated the licenses info and changelogs (for screen layouts)

![vocaluxe-2024-12-smartphone-control](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/20b6594d-e062-425f-bf2a-4590e7b12804)

marwin89 commented:
## 🚀 ready to merge
@flokuep: 
- For me this PullRequest is finished for now and can be merged. This is a big progress 😄🏁 
- We can always improve the logo or elements of the design (via other pullrequests or issues). 👍 
- If there are any errors let me know

jimmyhawkin commented:
Stefans idé is spot on What i was thinking about the logo aswell

flokuep reviewed (changes requested):
Thank you for working on that topic :) I've still two things that maybe need to be changed before merging:
- It looks like that the guest-profiles are deleted, for a better first-start-experience it would be good to have two or three default profiles. Or do you have a different opinion on that topic?
- The current theme should be removed and added to the GitHub Repository I've recently created for.

marwin89 commented:
@flokuep:
- I like the guest profiles and agree that they're usefull 👍 👍  But strangely I can't re-add them via github (it's not recognizing file changes in profiles folder 😕 . can you dismiss their deletion (in commit 1) in the first place anyhow?
- I got some troubles with moving genius. Can you put some dummy file in the genius-theme-repository? To upload files in it I need to fork that repository, but I can only fork it, if it's not empty 😄

marwin89 committed:
Move Genius to other repository

Vocaluxe 2024 design is the new default and main design for future developement. That's why Genius theme will be moved in a seperate repository where interested people can maintain it for next updates.

flokuep committed:
Create guest1.xml

flokuep committed:
Create guest2.xml

flokuep committed:
Create guest3.xml

flokuep commented:
Done :) I guess we just need to chose different default avatars now, but guest-files are back.

marwin89 commented:
![Screenshot 2024-01-01 132626](https://github.com/Vocaluxe/Vocaluxe/assets/63517874/4c8dd13f-6077-4b30-af1f-fae6da3782f7)
Default avatars for guests are already included. file names are the same as before. Vocaluxes crawls all folder levels. 👍

marwin89 commented:
@flokuep : Genius is successfully moved, can you approve the pull-req? ✔️
